### PR TITLE
Issue #162 Impl check_access

### LINKED ISSUE #162: Impl checkAccess
[JS reference implementation](https://github.com/AdExNetwork/adex-validator/blob/05-stable/services/sentry/lib/access.js)

### PR DESCRIPTION
**Pull request by elpiel**
Resolves #162

### CODE REVIEW

File: sentry/src/access.rs [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+    let current_time = Utc::now();
+
+    if current_time > channel.valid_until {
+        return Response::ChannelIsExpired;
+    }
+
+    if current_time > channel.spec.withdraw_period_start && !events.iter().all(is_close_event) {
+        return Response::ChannelIsPastWithdrawPeriod;
+    }
+
```
Comment (samparsky) [inline sentry/src/access.rs]: There is a missing check for not rate-limiting channel creator

File: sentry/src/access.rs [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+use crate::Session;
+
+#[derive(Debug, PartialEq, Eq)]
+pub enum Response {
+    OnlyCreatorCanCloseChannel,
+    ChannelIsExpired,
+    ChannelIsPastWithdrawPeriod,
+    RulesError(String),
+    Success,
+}
```
[the remark this answers is not in the document]
Comment (elpiel) [reply sentry/src/access.rs]: ReduceResult, Result? `Response` is too vague indeed.
Keep in mind it's also namespaced, so even if we have many `Response` or `Xxxx` structs it doesn't really matter.

File: sentry/src/access.rs (hunk 1 above)
Comment (elpiel) [reply sentry/src/access.rs]: 1 line bellow? Isn't this what you mean?
https://github.com/AdExNetwork/adex-validator/blob/05-stable/services/sentry/lib/access.js#L34

Comment (samparsky) [reply sentry/src/access.rs]: It's missing in `05-stable` but a channel creator shouldn't be rate-limited
https://github.com/AdExNetwork/adex-validator/blob/master/services/sentry/lib/access.js#L37

File: sentry/src/access.rs [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+
+// @TODO: Make pub(crate)
+pub async fn check_access(
+    redis: &SharedConnection,
+    session: &Session,
+    rate_limit: &RateLimit,
+    channel: &Channel,
+    events: &[Event],
+) -> Response {
```
Comment (samparsky) [inline sentry/src/access.rs]: `check_access` should return a Result, it provides a better API

File: sentry/src/access.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (samparsky) [reply sentry/src/access.rs]: In the JS impl we don't use the error types for anything, we mostly care if an error occurs
https://github.com/AdExNetwork/adex-validator/blob/05-stable/services/sentry/lib/access.js#L25

**Pull request merged**